File: exercism/go/difference-of-squares/difference_of_squares.go
package diffsquares

import "math"

func SquareOfSum(input int) int {
	total := 0
	for i := 1; i <= input; i++ {
		total += i
	}
	return int(math.Pow(float64(total), 2))
}

func SumOfSquares(input int) int {
	total := 0.0
	for i := 1; i <= input; i++ {
		total += math.Pow(float64(i), 2)
	}
	return int(total)
}

func Difference(input int) int {
	return SquareOfSum(input) - SumOfSquares(input)
}

File: exercism/kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(val s: Int) {

    fun sumOfSquares() = (1..s).map { Math.pow(it.toDouble(), 2.0) }.sum().toInt()
    fun squareOfSum() = Math.pow((1..s).sum().toDouble(), 2.0).toInt()

    fun difference() = squareOfSum() - sumOfSquares()
}
